src/MCPClient/lib/clientScripts/verify_AIP.py

```diff
@@ -0,0 +1,244 @@
+            expected_checksum = get_expected_checksum(
+                file_, sip_uuid, checksum_type, path2checksum, file_path,
+                manifest_path)
+            assert_checksums_match(file_, sip_uuid, checksum_type,
+                                   expected_checksum)
+            verification_count += 1
+    except VerifyChecksumsError as err:
+        print(err)
+        return 1
+    event_outcome_detail_note = (
+        'All {verification_count} checksums generated at start of transfer'
+        ' match those generated by BagIt (bag).'.format(
+            verification_count=verification_count))
+    write_premis_event(sip_uuid, checksum_type, 'Pass',
+                       event_outcome_detail_note)
+    print(event_outcome_detail_note)
+    return 0
+
+
+def verify_aip():
+    """Verify the AIP was bagged correctly by extracting it and running
+    verification on its contents. This is also where we verify the checksums
+    now that the verifyPREMISChecksums_v0.0 ("Verify checksums generated on
+    ingest") micro-service has been removed. It was removed because verifying
+    checksums by calculating them in that MS and then having bagit calculate
+    them here was redundant.
+
+    sys.argv[1] = UUID
+      UUID of the SIP, which will become the UUID of the AIP
+    sys.argv[2] = current location
+      Full absolute path to the AIP's current location on the local filesystem
+    """
+
+    sip_uuid = sys.argv[1]  # %sip_uuid%
+    aip_path = sys.argv[2]  # SIPDirectory%%sip_name%-%sip_uuid%.7z
+
+    temp_dir = mcpclient_settings.TEMP_DIRECTORY
+
+    is_uncompressed_aip = os.path.isdir(aip_path)
+
+    if is_uncompressed_aip:
+        bag = aip_path
+    else:
+        try:
+            extract_dir = os.path.join(temp_dir, sip_uuid)
+            bag = extract_aip(aip_path, extract_dir)
+        except Exception:
+            print('Error extracting AIP at "{}"'.format(aip_path), file=sys.stderr)
+            return 1
+
+    verification_commands = [
+        '/usr/share/bagit/bin/bag verifyvalid "{}"'.format(bag),
+        '/usr/share/bagit/bin/bag checkpayloadoxum "{}"'.format(bag),
+        '/usr/share/bagit/bin/bag verifycomplete "{}"'.format(bag),
+        '/usr/share/bagit/bin/bag verifypayloadmanifests "{}"'.format(bag),
+        '/usr/share/bagit/bin/bag verifytagmanifests "{}"'.format(bag),
+    ]
+    return_code = 0
+    for command in verification_commands:
+        print("Running test: ", command)
+        exit_code, _, _ = executeOrRun("command", command, printing=True)
+        if exit_code != 0:
+            print("Failed test: ", command, file=sys.stderr)
+            return_code = 1
+
+    if return_code == 0:
+        return_code = verify_checksums(bag, sip_uuid)
+    else:
+        print('Not verifying checksums because other tests have already'
+              ' failed.')
+
+    # cleanup
+    if not is_uncompressed_aip:
+        shutil.rmtree(extract_dir)
```

Comment:
Do we want to make the script report an error if the clean up did not succeed? You may want to trap `OSError` if that's not the case.

Reply:
sounds good